samples/same54xprov2/basic-sample/src/iotconnect_app.c

```diff
@@ -0,0 +1,310 @@
+//
+// Copyright: Avnet 2021
+// Created by Nik Markovic <nikola.markovic@avnet.com> on 4/19/21.
+//
+
+#include "app_config.h"
+
+#include "nx_api.h"
+#include "nxd_dns.h"
+#include "iotconnect_common.h"
+#include "iotconnect.h"
+#include "iotc_auth_driver.h"
+#include "sw_auth_driver.h"
+#include "weather.h"
+
+#ifdef ENABLE_DDIM_PKCS11_ATCA_DRIVER_SAMPLE
+#include "pkcs11_atca_auth_driver.h"
+#include "atcacert/atcacert_pem.h"
+#endif
+
+extern UCHAR _nx_driver_hardware_address[];
+static IotConnectAzrtosConfig azrtos_config;
+static IotcAuthInterfaceContext auth_driver_context = NULL;
+
+#ifdef ENABLE_DDIM_PKCS11_ATCA_DRIVER_SAMPLE
+static char duid_buffer[IOTC_COMMON_NAME_MAX_LEN]; // from ATECC608 common name
+#endif
+#define APP_VERSION "01.00.00"
+
+//#define MEMORY_TEST
```

Comment:
Remove this and pass it in via -DMEMORY_TEST if wanting to build the memory test would be a tiny bit tidier